types/src/eth.rs

```diff
@@ -25,6 +26,36 @@ pub struct BlockResult {
     pub mpt_witness: Vec<SMTTrace>,
 }
 
+impl From<BlockTrace> for EthBlock {
+    fn from(mut b: BlockTrace) -> Self {
+        let mut txs = Vec::new();
+        for (idx, tx_data) in b.transactions.iter_mut().enumerate() {
+            let from = tx_data.transaction.recover_from().unwrap();
+            let tx = Transaction {
+                from,
+                block_hash: b.header.hash,
+                block_number: b.header.number,
+                transaction_index: Some(U64::from(idx)),
+                ..tx_data.clone().transaction
+            };
+            txs.push(tx)
+        }
+        EthBlock {
+            transactions: txs,
+            ..b.header
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Default, Debug, Clone)]
+pub struct TransactionData {
+    #[serde(rename = "isCreate", default)]
+    pub is_create: bool,
+    pub from: Option<Address>,
+    #[serde(flatten)]
+    pub transaction: Transaction,
+}
```

Comment:
we can replace this whole struct with `Transaction` now

Reply:
Do we need to change the l2geth first?